src/installer/managed/Microsoft.NET.HostModel/MachO/RequirementsBlob.cs

```diff
@@ -0,0 +1,31 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.NET.HostModel.MachO;
+
+[StructLayout(LayoutKind.Sequential)]
+internal struct RequirementsBlob
+{
+    private BlobMagic _magic;
+    private uint _requirementsBlobSize;
+    private uint _entitlementsBlobLength;
+
+    public static RequirementsBlob Empty = new RequirementsBlob
+    {
+        _magic = (BlobMagic)((uint)BlobMagic.Requirements).MakeBigEndian(),
+        _requirementsBlobSize = 12u.MakeBigEndian(),
+        _entitlementsBlobLength = 0
+    };
+
+    public byte[] GetBytes()
+    {
+        byte[] buffer = new byte[12];
+        BitConverter.GetBytes((uint)_magic).CopyTo(buffer, 0);
+        BitConverter.GetBytes(_requirementsBlobSize).CopyTo(buffer, 4);
+        BitConverter.GetBytes(_entitlementsBlobLength).CopyTo(buffer, 8);
```

Comment:
Nit: cast the span and write. Array allocation should satisfy the alignment for uint32.

Reply:
Not sure I follow what you mean here, could you elaborate?

Reply:
`BinaryPrimitives.WriteInt32BigEndian` etc, which should be available for netstandard2.0 . `BitConverte` is endianness-dependent, so I'm not sure which way is correct.